Competition2018/src/test/java/competition/subsystems/climberdeploy/Command/Tests/DetractClimberArmCommandTest.java

```diff
@@ -0,0 +1,39 @@
+package competition.subsystems.climberdeploy.Command.Tests;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+import competition.subsystems.climberdeploy.ClimberdeploySubsystem;
+import competition.subsystems.climberdeploy.commands.DetractClimberArmCommand;
+import xbot.common.controls.actuators.mock_adapters.MockCANTalon;
+import xbot.common.injection.BaseWPITest;
+
+public class DetractClimberArmCommandTest extends BaseWPITest{
+	
+	ClimberdeploySubsystem deploy;
+	DetractClimberArmCommand command;
+	
+	@Override
+	public void setUp() {
+		super.setUp();
+		deploy = injector.getInstance(ClimberdeploySubsystem.class);
+		command = injector.getInstance(DetractClimberArmCommand.class);
+		deploy.temporaryHack();
+	}
+	
+	@Test
+	public void simpleTest() {
+		command.initialize();
+		command.execute();
+	}
+	
+	@Test
+	public void checkDetractClimberArm() {
```

Comment:
To be sure your test works, you need to assert that the speed beforehand was 0. Otherwise, what if it was -0.2 all along? =]

Reply:
I have tried to fix this by adding a check for zero speed between initialize and execute. But I need further clarification relating to the question and review of my proposed solution.